index.js

```diff
@@ -13,22 +13,34 @@ function fastifyPostgres (fastify, options, next) {
     }
   }
 
-  const pool = new pg.Pool(options)
+  const name = options.name
+  delete options.name
 
-  fastify.decorate('pg', {
+  const pool = new pg.Pool(options)
+  const db = {
     connect: pool.connect.bind(pool),
     pool: pool,
     Client: pg.Client,
     query: pool.query.bind(pool)
-  })
+  }
 
-  fastify.addHook('onClose', onClose)
+  if (name) {
+    if (!fastify.pg) {
+      fastify.decorate('pg', {})
+    }
 
-  next()
-}
+    fastify.pg[name] = db
+  } else {
+    if (fastify.pg) {
+      console.warn('fastify-postgres has already registered')
```

Comment:
we should error if it was already register.

Reply:
got it